mage_ai/frontend/pages/pipelines/index.tsx

```diff
@@ -1204,6 +1239,26 @@ function PipelineListPage() {
             }),
             uuid: 'clone',
           },
+          {
+            label: () => 'Download (keep folder structure)',
+            onClick: () => {
+              downloadPipeline({
+                pipelineUUID: selectedPipeline?.uuid,
+                filesOnly: false
+              });
+            },
+            uuid: 'download',
+          },
+          {
+            label: () => 'Download (without folder structure)',
+            onClick: () => {
+              downloadPipeline({
```

Comment:
Missing dependency `downloadPipeline` in useCallback dependency array here: https://github.com/mage-ai/mage-ai/pull/3813/files#diff-22d0517fd466848fa8e87718671265bcaecd728c9a6a4407b128c3532a88c6d9R1479-R1494